### PR TITLE
Removed `GetAwaiter().GetResult()` calls from `JobRunShell`

### PR DESCRIPTION
**Pull request by danielmarbach**
Connects to #331 

Gets rid of the blocking calls in the `JobRunShell`

Based on #339, will rebase when merged.

### CODE REVIEW

**Comment by danielmarbach** [comment]
Rebased and force pushed

File: src/Quartz/Core/QuartzScheduler.cs [hunk 1]
```diff
@@ -2004,20 +2001,16 @@ public virtual Task NotifySchedulerListenersJobDeletedAsync(JobKey jobKey)
...
-                var jobDetail = jec.JobDetail;
+                var jobDetail = cancellableJobExecutionContext.JobDetail;
                 if (jobKey.Equals(jobDetail.Key))
                 {
-                    // set botch cancellation token and interrupt if interface is implemented
-                    jec.Cancel();
-                    IInterruptableJob jobInstance = jec.JobInstance as IInterruptableJob;
-                    jobInstance?.Interrupt();
+                    cancellableJobExecutionContext.Cancel();
                     interrupted = true;
```
Comment (lahma) [inline src/Quartz/Core/QuartzScheduler.cs]: probably should break the loop here

File: src/Quartz/Core/QuartzScheduler.cs [hunk 2]
```diff
@@ -2038,22 +2031,20 @@ public virtual Task<bool> InterruptAsync(JobKey jobKey)
...
-                if (jec.FireInstanceId.Equals(fireInstanceId))
+                if (cancellableJobExecutionContext.FireInstanceId.Equals(fireInstanceId))
                 {
-                    // set botch cancellation token and interrupt if interface is implemented
-                    jec.Cancel();
-                    IInterruptableJob jobInstance = jec.JobInstance as IInterruptableJob;
-                    jobInstance?.Interrupt();
-                    return Task.FromResult(true);
+                    cancellableJobExecutionContext.Cancel();
+                    interrupted = true;
```
Comment (lahma) [inline src/Quartz/Core/QuartzScheduler.cs]: and break the loop here too

**Comment by danielmarbach** [comment]
Fixed. Rebased. Pushed

**Comment by lahma** [comment]
Thank you!

**Pull request merged**